<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Gota Morota" />

<meta name="date" content="2018-11-12" />

<title>Introduction to Quantitative Genetics</title>

<script src="lab1_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="lab1_files/bootstrap-3.3.6/css/bootstrap.min.css" rel="stylesheet" />
<script src="lab1_files/bootstrap-3.3.6/js/bootstrap.min.js"></script>
<script src="lab1_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<script src="lab1_files/navigation-1.1/tabsets.js"></script>
<script src="lab1_files/navigation-1.1/codefolding.js"></script>
<link href="lab1_files/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
<script src="lab1_files/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
<link href="lab1_files/readthedown-0.1/readthedown.css" rel="stylesheet" />
<script src="lab1_files/readthedown-0.1/readthedown.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #ffffff; color: #1f1c1b; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; background-color: #ffffff; color: #a0a0a0; border-right: 1px solid #a0a0a0; }
td.sourceCode { padding-left: 5px; }
pre, code { color: #1f1c1b; background-color: #ffffff; }
code > span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code > span.dt { color: #0057ae; } /* DataType */
code > span.dv { color: #b08000; } /* DecVal */
code > span.bn { color: #b08000; } /* BaseN */
code > span.fl { color: #b08000; } /* Float */
code > span.cn { color: #aa5500; } /* Constant */
code > span.ch { color: #924c9d; } /* Char */
code > span.sc { color: #3daee9; } /* SpecialChar */
code > span.st { color: #bf0303; } /* String */
code > span.vs { color: #bf0303; } /* VerbatimString */
code > span.ss { color: #ff5500; } /* SpecialString */
code > span.im { color: #ff5500; } /* Import */
code > span.co { color: #898887; } /* Comment */
code > span.do { color: #607880; } /* Documentation */
code > span.an { color: #ca60ca; } /* Annotation */
code > span.cv { color: #0095ff; } /* CommentVar */
code > span.ot { color: #006e28; } /* Other */
code > span.fu { color: #644a9b; } /* Function */
code > span.va { color: #0057ae; } /* Variable */
code > span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code > span.op { color: #1f1c1b; } /* Operator */
code > span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code > span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code > span.pp { color: #006e28; } /* Preprocessor */
code > span.at { color: #0057ae; } /* Attribute */
code > span.re { color: #0057ae; } /* RegionMarker */
code > span.in { color: #b08000; } /* Information */
code > span.wa { color: #bf0303; } /* Warning */
code > span.al { color: #bf0303; font-weight: bold; } /* Alert */
code > span.er { color: #bf0303; text-decoration: underline; } /* Error */
code > span. { color: #1f1c1b; } /* Normal */
</style>


</head>

<body>


<div id="content" data-toggle="wy-nav-shift">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<nav id="nav-top" role="navigation" aria-label="top navigation">
    <a role="button" href="#" data-toggle="wy-nav-top"><span class="glyphicon glyphicon-menu-hamburger"></span></a>
</nav>

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span class="text-muted">Code</span> <span class="text-muted caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All</a></li>
</ul>
</div>

<div id="header">
<h1 class="title">Introduction to Quantitative Genetics</h1>
<h1 class="subtitle">Statistical Methods for Omics-assisted Breeding Hands-on Workshop</h1>
</div>


<div id="table-of-contents">
    <h2><a href="#content">Introduction to Quantitative Genetics</a></h2>
    <div id="text-table-of-contents">
      <ul>
      <li><a href="#data">Data</a></li>
      <li><a href="#genotypic-frequency">Genotypic frequency</a></li>
      <li><a href="#another-approach-for-obtaining-allele-frequency">Another approach for obtaining allele frequency</a></li>
      <li><a href="#compute-allele-frequencies-for-all-snps">Compute allele frequencies for all SNPs</a></li>
      <li><a href="#minor-allele-frequency">Minor allele frequency</a></li>
      <li><a href="#expectations-and-variances">Expectations and variances</a></li>
      </ul>
    </div>
</div>

<div id="main">
<div id="data" class="section level2">
<h2>Data</h2>
<p>We are going to use the <code>mice</code> data included in the <a href="https://cran.r-project.org/web/packages/BGLR/index.html">BGLR</a> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BGLR)
<span class="kw">data</span>(mice)
W &lt;-<span class="st"> </span>mice.X[, <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>]</code></pre></div>
<p>We will learn how to compute allele and genotypic frequencies in R using the cattle data set.</p>
<p>Let’s compute the allele frequency of the first SNP. The <code>table</code> function returns frequncies of genotypes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(W[, <span class="dv">1</span>])</code></pre></div>
<p>We can see that there are 541 <em>AA</em> animals, 929 <em>Aa</em> animals, and 344 <em>aa</em> animals. Let’s assign these numbers into variables.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nAA &lt;-<span class="st"> </span><span class="kw">table</span>(W[, <span class="dv">1</span>])[<span class="dv">3</span>]
nAa &lt;-<span class="st"> </span><span class="kw">table</span>(W[, <span class="dv">1</span>])[<span class="dv">2</span>]
naa &lt;-<span class="st"> </span><span class="kw">table</span>(W[, <span class="dv">1</span>])[<span class="dv">1</span>]</code></pre></div>
<p>Allele frequency of <em>A</em> is given by <span class="math display">\[ 
f(A) = p = \frac{2 \times (\text{no. of } AA \text{ individuals}) + 1 \times (\text{no. of } Aa \text{ individuals})}{2 \times \text{total no. of individuals}}.
\]</span></p>
<div id="exercise-1" class="section level3">
<h3>Exercise 1</h3>
<p>Use the variables <code>nAA</code>, <code>nAa</code>, and <code>naa</code> defined above and compute the allele frequencies of <em>A</em> and <em>a</em> in the first SNP.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>nAA <span class="op">+</span><span class="st"> </span><span class="dv">1</span> <span class="op">*</span><span class="st"> </span>nAa)<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>(nAA <span class="op">+</span><span class="st"> </span>nAa <span class="op">+</span><span class="st"> </span>naa))
p
q &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>p
q</code></pre></div>
</div>
</div>
<div id="genotypic-frequency" class="section level2">
<h2>Genotypic frequency</h2>
<p>Genotypic frequency is given by <span class="math display">\[ 
f(AA) = P = \frac{\text{No. of } AA \text{ individuals}}{\text{Total no. individuals}} \\
f(Aa) = H = \frac{\text{No. of } Aa \text{ individuals}}{\text{Total no. individuals}} \\
f(aa) = Q = \frac{\text{No. of } aa \text{ individuals}}{\text{Total no. individuals}}. \\
\]</span></p>
<div id="exercise-2" class="section level3">
<h3>Exercise 2</h3>
<p>What are the genotypic frequencies of <code>AA</code>, <code>Aa</code>, and <code>aa</code> in the first SNP?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">P &lt;-<span class="st"> </span>(nAA)<span class="op">/</span>(nAA <span class="op">+</span><span class="st"> </span>nAa <span class="op">+</span><span class="st"> </span>naa)
P
H &lt;-<span class="st"> </span>(nAa)<span class="op">/</span>(nAA <span class="op">+</span><span class="st"> </span>nAa <span class="op">+</span><span class="st"> </span>naa)
H
Q &lt;-<span class="st"> </span>(naa)<span class="op">/</span>(nAA <span class="op">+</span><span class="st"> </span>nAa <span class="op">+</span><span class="st"> </span>naa)
Q</code></pre></div>
</div>
</div>
<div id="another-approach-for-obtaining-allele-frequency" class="section level2">
<h2>Another approach for obtaining allele frequency</h2>
<p><span class="math display">\[ 
f(A) = p = \frac{2 \times (\text{frequency of } AA) + 1 \times (\text{frequency of } Aa)}{2 \times (\text{frequency of } AA + Aa + aa)}.
\]</span></p>
<div id="exercise-3" class="section level3">
<h3>Exercise 3</h3>
<p>Use the variables <code>P</code>, <code>H</code>, and <code>Q</code> defined above and compute the allele frequencies of <em>A</em> and <em>a</em> in the first SNP.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p.ex3 &lt;-<span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>P <span class="op">+</span><span class="st"> </span><span class="dv">1</span> <span class="op">*</span><span class="st"> </span>H)<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>(P <span class="op">+</span><span class="st"> </span>H <span class="op">+</span><span class="st"> </span>Q))
p.ex3
q.ex3 &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>p.ex3
q.ex3</code></pre></div>
</div>
<div id="exercise-4" class="section level3">
<h3>Exercise 4</h3>
<p>What are the genotypic frequencies of <code>AA</code>, <code>Aa</code>, and <code>aa</code> in the second SNP?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nAA &lt;-<span class="st"> </span><span class="kw">table</span>(W[, <span class="dv">2</span>])[<span class="dv">3</span>]
nAa &lt;-<span class="st"> </span><span class="kw">table</span>(W[, <span class="dv">2</span>])[<span class="dv">2</span>]
naa &lt;-<span class="st"> </span><span class="kw">table</span>(W[, <span class="dv">2</span>])[<span class="dv">1</span>]

p &lt;-<span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>nAA <span class="op">+</span><span class="st"> </span><span class="dv">1</span> <span class="op">*</span><span class="st"> </span>nAa)<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>(nAA <span class="op">+</span><span class="st"> </span>nAa <span class="op">+</span><span class="st"> </span>naa))
p
q &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>p
q

P &lt;-<span class="st"> </span>(nAA)<span class="op">/</span>(nAA <span class="op">+</span><span class="st"> </span>nAa <span class="op">+</span><span class="st"> </span>naa)
P
H &lt;-<span class="st"> </span>(nAa)<span class="op">/</span>(nAA <span class="op">+</span><span class="st"> </span>nAa <span class="op">+</span><span class="st"> </span>naa)
H
Q &lt;-<span class="st"> </span>(naa)<span class="op">/</span>(nAA <span class="op">+</span><span class="st"> </span>nAa <span class="op">+</span><span class="st"> </span>naa)
Q</code></pre></div>
</div>
</div>
<div id="compute-allele-frequencies-for-all-snps" class="section level2">
<h2>Compute allele frequencies for all SNPs</h2>
<p>So far we have learned how to compute the allele frequency of a single SNP using the <code>table</code> function. Here, we consider how to compute allele frequencies for the entire SNPs. Of course we can apply the <code>table</code> function manually one at a time. However, this approach takes too much time to compute allele frequencies for 6,960 SNPs. Recall that allele frequency of <em>A</em> is given by <span class="math display">\[ 
f(A) = p = \frac{2 \times (\text{no. of } AA \text{ individuals}) + 1 \times (\text{no. of } Aa \text{ individuals})}{2 \times \text{total no. of individuals}}.
\]</span> We can rewrite this equation into <span class="math display">\[ 
f(A) = p = \frac{(\text{no. of } A \text{ allele in the population})}{2 \times \text{total no. of individuals}}.
\]</span> This suggests that all we need is the number of <span class="math inline">\(A\)</span> allele or reference allele <span class="math inline">\(a\)</span> for each SNP. The <code>sum</code> function returns the number of reference allele <span class="math inline">\(A\)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(W[, <span class="dv">1</span>])  <span class="co"># sum of A allele in the first SNP</span>
<span class="kw">sum</span>(W[, <span class="dv">2</span>])  <span class="co"># sum of A allele in the second SNP</span></code></pre></div>
<p>How to repeat this operation for 300 SNPs? The <code>colSums</code> function returns the sum of each column in a matrix as a vector.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colSums</span>(W)</code></pre></div>
<p>Note that <code>colSums(W)</code> gives the numerator of the above equation. We then divide this number by <span class="math inline">\(2 \times \text{total no. of individuals}\)</span>. The function <code>nrows</code> returns the number of rows.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">colSums</span>(W)<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(W))</code></pre></div>
<p>The variable <code>p</code> is a vector and it contains the allele frequencies of reference allele for 6,960 SNPs.</p>
<div id="exercise-5" class="section level3">
<h3>Exercise 5</h3>
<p>What is the allele frequency of reference allele in the 50th SNP?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p[<span class="dv">50</span>]</code></pre></div>
</div>
</div>
<div id="minor-allele-frequency" class="section level2">
<h2>Minor allele frequency</h2>
<p>In most cases, people report a minor allele frequency, which is the frequency of less frequent allele in a given SNP. We can convert allele frequencies into minor allele frquencies by using the <code>ifelse</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maf &lt;-<span class="st"> </span><span class="kw">ifelse</span>(p <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>, <span class="dv">1</span> <span class="op">-</span><span class="st"> </span>p, p)</code></pre></div>
<div id="exercise-6" class="section level3">
<h3>Exercise 6</h3>
<p>What is the minor allele frquency of reference allele in the 30th SNP?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maf[<span class="dv">30</span>]</code></pre></div>
</div>
<div id="exercise-7" class="section level3">
<h3>Exercise 7</h3>
<p>What is the mean of minor allele frquencies?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mean</span>(maf)</code></pre></div>
</div>
</div>
<div id="expectations-and-variances" class="section level2">
<h2>Expectations and variances</h2>
<div id="exercise-8" class="section level3">
<h3>Exercise 8</h3>
<p>Compute the allele frequency of SNP markers. Recall that the expectation of genotype, <span class="math inline">\(E(W)\)</span>, is given by <span class="math inline">\(2p\)</span>, where <span class="math inline">\(p\)</span> is the frequency of reference allele. Verify that <span class="math inline">\(2p\)</span> is equal to the mean of each genotype obtained from the <code>colMeans()</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p3 &lt;-<span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">colSums</span>(W <span class="op">==</span><span class="st"> </span><span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="dv">1</span> <span class="op">*</span><span class="st"> </span><span class="kw">colSums</span>(W <span class="op">==</span><span class="st"> </span><span class="dv">1</span>))<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(W))
<span class="kw">table</span>(<span class="kw">colMeans</span>(W) <span class="op">==</span><span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>p3)

p &lt;-<span class="st"> </span><span class="kw">colSums</span>(W)<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(W))
<span class="kw">table</span>(<span class="kw">colMeans</span>(W) <span class="op">==</span><span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>p)

p2 &lt;-<span class="st"> </span><span class="kw">colMeans</span>(W)<span class="op">/</span><span class="dv">2</span>
<span class="kw">table</span>(p <span class="op">==</span><span class="st"> </span>p2)</code></pre></div>
</div>
<div id="exercise-9" class="section level3">
<h3>Exercise 9</h3>
<p>Recall that the variance of genotype, <span class="math inline">\(Var(W)\)</span>, is given by <span class="math inline">\(2p(1-p)\)</span>. Verify that <span class="math inline">\(2p(1-p)\)</span> is close to the variance of each genotype obtained from the <code>var()</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">var1 &lt;-<span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>p <span class="op">*</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>p)
var2 &lt;-<span class="st"> </span><span class="kw">diag</span>(<span class="kw">var</span>(W))
var2 &lt;-<span class="st"> </span><span class="kw">apply</span>(W, <span class="dv">2</span>, var)
<span class="kw">table</span>(var1 <span class="op">==</span><span class="st"> </span>var2)
<span class="kw">head</span>(var1)
<span class="kw">head</span>(var2)
<span class="kw">all.equal</span>(var1, var2, <span class="dt">tol =</span> <span class="fl">0.05</span>)</code></pre></div>
</div>
<div id="exercise-10" class="section level3">
<h3>Exercise 10</h3>
<p>Create a new marker matrix <code>X</code> from <code>W</code> and recode markers so that three genotypes <span class="math inline">\(AA\)</span>, <span class="math inline">\(Aa\)</span>, and <span class="math inline">\(aa\)</span> are coded as 1, 0, and -1, respectively.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">X &lt;-<span class="st"> </span>W <span class="op">-</span><span class="st"> </span><span class="dv">1</span></code></pre></div>
<p>Recall that the expectation of genotype, <span class="math inline">\(E(X)\)</span>, is given by <span class="math inline">\(2p-1\)</span>, where <span class="math inline">\(p\)</span> is the frequency of reference allele. Verify that <span class="math inline">\(2p-1\)</span> is equal to the mean of each genotype obtained from the <code>colMeans()</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2 &lt;-<span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">colSums</span>(X <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="dv">1</span> <span class="op">*</span><span class="st"> </span><span class="kw">colSums</span>(X <span class="op">==</span><span class="st"> </span><span class="dv">0</span>))<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(X))
<span class="kw">table</span>(p <span class="op">==</span><span class="st"> </span>p2)
<span class="kw">all.equal</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>p <span class="op">-</span><span class="st"> </span><span class="dv">1</span>, <span class="kw">colMeans</span>(X), <span class="dt">tol =</span> <span class="fl">0.005</span>)</code></pre></div>
</div>
<div id="exercise-12" class="section level3">
<h3>Exercise 12</h3>
<p>Recall that the variance of genotype, <span class="math inline">\(Var(X)\)</span>, remains the same and is given by <span class="math inline">\(2p(1-p)\)</span>. Verify that <span class="math inline">\(2p(1-p)\)</span> is close to the variance of each genotype obtained from the <code>var()</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">var1 &lt;-<span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>p <span class="op">*</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>p)
var2 &lt;-<span class="st"> </span><span class="kw">var</span>(X)
var2 &lt;-<span class="st"> </span><span class="kw">apply</span>(X, <span class="dv">2</span>, var)
<span class="kw">all.equal</span>(var1, var2, <span class="dt">tol =</span> <span class="fl">0.05</span>)</code></pre></div>
</div>
<div id="exercise-13" class="section level3">
<h3>Exercise 13</h3>
<p>Verify that no matter how we code markers, centered marker codes, <span class="math inline">\(W - E(W)\)</span> and <span class="math inline">\(X - E(X)\)</span>, remain the same.</p>
</div>
<div id="exercise-14" class="section level3">
<h3>Exercise 14</h3>
<p>We will recode the SNP genotypes so that now the major allele is treated as a reference allele. Store the new coding into the <code>W2</code> variable.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">W2 &lt;-<span class="st"> </span>W
W2[W2 <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="dv">3</span>
W2[W2 <span class="op">==</span><span class="st"> </span><span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">0</span>
W2[W2 <span class="op">==</span><span class="st"> </span><span class="dv">3</span>] &lt;-<span class="st"> </span><span class="dv">2</span></code></pre></div>
<p>Compute the allele freqeuncy of SNP markers using <code>W2</code>. Compare your result with the allele frequency obtained from <code>W</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Freq of A (W)</span>
p &lt;-<span class="st"> </span><span class="kw">colSums</span>(W)<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(W))
<span class="co"># Freq of A (W2)</span>
pW2 &lt;-<span class="st"> </span><span class="kw">colSums</span>(W2)<span class="op">/</span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(W2))

<span class="kw">head</span>(p)
<span class="kw">head</span>(pW2)


<span class="co"># SNP 1</span>
<span class="kw">table</span>(W[, <span class="dv">1</span>])
<span class="kw">table</span>(W2[, <span class="dv">1</span>])

<span class="co"># SNP 16</span>
<span class="kw">table</span>(W[, <span class="dv">16</span>])
<span class="kw">table</span>(W2[, <span class="dv">16</span>])</code></pre></div>
</div>
</div>
</div>


</div>

<div id="postamble" data-toggle="wy-nav-shift" class="status">
<p class="author"><span class="glyphicon glyphicon-user"></span> Gota Morota</p>
<p class="date"><span class="glyphicon glyphicon-calendar"></span> November 12, 2018</p>
</div>


<script>
$(document).ready(function () {
 	  });
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
